SBC centered metrics

lindsay-fredrick opened the pull request:
Some exploratory code to:
* Look at SBC centered metrics (instead of region centered metrics) and visuals.

The main changes are all incorporated into `03b-sbc-centric-stats.R`, although I did add a couple of plot functions to the `plots.R` script, as well as some more provincial level plots to `02b-plot-csds.R`. 

Outputs include 3 tables for SBC location centered metrics (one with driving metrics, one with population metrics, and one with CSD counts). There are also some plots: driving distance histograms, population pyramids, and maps of the 'catchments' (colored by DB driving time) for each of the SBC locations we considered. Provincial level plots show where the catchments are.

BonnieJRobert commented on R/03-sbc-centric-stats.R:
```diff
@@ -0,0 +1,535 @@
+    height = 12,
+    dpi = 300
+  )
+}
```
There are a few of these calculations you could make more robust.
```suggestion
    est_pop = sum(population, na.rm = TRUE),
```

lindsay-fredrick replied:
fixed in new file

BonnieJRobert commented on R/03-sbc-centric-stats.R:
```diff
@@ -0,0 +1,535 @@
+drive_measures <- drivetime_data_reduced |>
+  group_by(assigned) |>
+  summarize(
+    n_addresses = n(),
```
Consider adding a few na.rm = TRUE in these summary calculations, esp. given the previous left joins

lindsay-fredrick replied:
fixed in new file

BonnieJRobert commented on R/03-sbc-centric-stats.R:
```diff
@@ -0,0 +1,535 @@
+  group_by(csd_clean) |>
+  mutate(csd_population = sum(population)) |>
+  ungroup() |>
+  mutate(pct_of_csd = population / csd_population)
```
```suggestion
  mutate(pct_of_csd = if_else(csd_population == 0, 0, population / csd_population))
```

lindsay-fredrick replied:
fixed in new file

BonnieJRobert commented on R/03-sbc-centric-stats.R:
```diff
@@ -0,0 +1,535 @@
+# now get pct of each 'clean' csd that is taken up by each DB
+prop_of_csd <- get_clean_csd |>
+  group_by(csd_clean) |>
+  mutate(csd_population = sum(population)) |>
```
```suggestion
  mutate(csd_population = sum(population, rm.na = TRUE)) |>
```

lindsay-fredrick replied:
fixed in new file

lindsay-fredrick commented:
@BonnieJRobert   I’m done with my re-factor: I addressed the comments you had already made while I was going through (and added in more na.rm=T where I could find the need!). It’s mostly still the same otherwise, other than a new piece of logic to assign unassigned DBs to SBC catchments. This is all plunked into the new `03a-create-catchments.R` file. I also updated the readme with the basic methodology of the 3 scripts which I hope will help with understanding/getting the report fleshed out!

lindsay-fredrick commented:
@BK01 if you want to have a look at this please do! Hopefully you can get the raw data sorted out on your end so that you can run all the files. Let us know if you're having R annoyances and we should be able to sort it out!

BonnieJRobert commented on R/03a-create-catchments.R:
```diff
@@ -0,0 +1,164 @@
+# QA the unassigned DBs 
+# save the list of unassigned dbs, together with their 2021 census pops
+db_check <- complete_assignments %>% 
+  left_join(pop_db, by='dbid') %>% 
```
pop_db hasn't been defined in this script yet

lindsay-fredrick replied:
updated!

BonnieJRobert commented on R/03b-plot-catchments.R:
```diff
@@ -0,0 +1,499 @@
+
+# Create original catchment areas (with unassigned areas)
+original_catchments <- db_shapefile |>
+  left_join(complete_assignments, by = "dbid") |>
```
I'm receiving this error after attempting to join: 

`! Can't join `x$dbid` with `y$dbid` due to incompatible types.
ℹ `x$dbid` is a <character>.
ℹ `y$dbid` is a <double>.`

BonnieJRobert commented on R/03b-plot-catchments.R:
```diff
@@ -0,0 +1,499 @@
+  mutate(across(c(landarea), as.numeric))
+
+# Read the pre-computed assignment data 
+complete_assignments <- read_csv(glue("{SRC_DATA_FOLDER}/complete_db_assignments.csv"))
```
```suggestion
complete_assignments <- 
  read_csv(glue("{SRC_DATA_FOLDER}/complete_db_assignments.csv")) %>%
  mutate(across(c(dbid, assigned, assignment_method), as.character))
```

BonnieJRobert commented on R/03b-plot-catchments.R:
```diff
@@ -0,0 +1,499 @@
+  ) +
+  # Colors for catchment areas
+  scale_fill_viridis_d(
+    option = "turbo",
```
Are you attached to _turbo_ or do you mind if we use the _mako_ (or _vridis_) option?  SBC locations pop with _mako_ in particular, but they are both less aggressive schemes and so easier to look at.

lindsay-fredrick replied:
I am not attached! Will update to mako :)

lindsay-fredrick replied:
Actually after looking at the mako map, I find it harder to distinguish between the different 'catchments', especially on the full province map. I Like the overall look of it better (with the yellow popping better you're right), but it (imo) does blend the catchment colors together a little bit too much.

BonnieJRobert commented on R/03c-sbc-centric-stats.R:
```diff
@@ -0,0 +1,570 @@
+}
+
+# Generate a map for each Service BC location
+for (loc_id in unique(map_data$assigned)) {
```
```suggestion
for (loc_id %in% unique(map_data$assigned)) {
```

lindsay-fredrick replied:
Leaving as is, no `%` for a for loop

BonnieJRobert commented on R/03c-sbc-centric-stats.R:
```diff
@@ -0,0 +1,570 @@
+
+# Create and save histograms for each facility
+facilities <- unique(drivetime_data_reduced$assigned)
+for (facility in facilities) {
```
```suggestion
for (facility %in% facilities) {
```

BonnieJRobert commented:
@lindsay-fredrick this looks good!  A few minor fixes needed but otherwise the code runs to completion.  I didn't run `ggsave` or `write_csv` lines (with one or two exceptions).

BonnieJRobert reviewed (changes requested):
@lindsay-fredrick this looks good!  A few minor fixes needed but otherwise the code runs to completion.  I didn't run `ggsave` or `write_csv` lines (with one or two exceptions).

lindsay-fredrick commented:
@BonnieJRobert I think I've addressed all your comments - I'm just going to merge main into this branch to deal with the conflicts. Let me know if it's good otherwise - if we're good I will merge after I resolve the conflicts!

BonnieJRobert commented:
> @BonnieJRobert I think I've addressed all your comments - I'm just going to merge main into this branch to deal with the conflicts. Let me know if it's good otherwise - if we're good I will merge after I resolve the conflicts!

Good to merge!
